src/main/java/ch/njol/skript/util/slot/SlotWithIndex.java

```diff
@@ -28,6 +28,12 @@ public abstract class SlotWithIndex extends Slot {
 	 * @return Index of the slot.
 	 */
 	public abstract int getIndex();
+
+	/**
+	 * Gets the raw index of this slot.
+	 * @return Raw index of the slot.
+	 */
+	public abstract int getRawIndex();
```

Comment:
IMO instead of requiring extending classes to implement this, you should do the following:
```suggestion
	public int getRawIndex() {
		return getIndex();
	}
```
If you do this, you can remove the changes made to EquipmentSlot too.